File: 1748-best-team-with-no-conflicts/1748-best-team-with-no-conflicts.cpp
class Solution {
public:
    int bestTeamScore(vector<int>& scores, vector<int>& ages) {
        int n = scores.size();
        vector<pair<int, int>> p(n);
        vector<int> ans(n);
        for(int i = 0; i < n; i++){
            p[i] = {ages[i], scores[i]};
        }
        sort(p.begin(), p.end());
        for(int i = 0; i < n; i++){
            ans[i] = p[i].second;
            for(int j = 0; j < i; j++){
                if(p[i].second >= p[j].second){
                    ans[i] = max(ans[i], ans[j] + p[i].second);
                }
            }
        }
        return *max_element(ans.begin(), ans.end());
    }
};

File: 1748-best-team-with-no-conflicts/1748-best-team-with-no-conflicts.py
class Solution:
    def bestTeamScore(self, scores: List[int], ages: List[int]) -> int:
        n = len(scores)
        p = sorted(zip(scores, ages), key=lambda x: (x[1], x[0]))
        dp = [0] * n
        
        for i in range(n):
            dp[i] = p[i][0]
            for j in range(i):
                if p[i][0] >= p[j][0]:
                    dp[i] = max(dp[i], dp[j] + p[i][0])
        return max(dp)



